src/core/tests/type_prop/broadcast.cpp

```diff
@@ -915,4 +919,308 @@ TEST(type_prop, broadcast_i32_shape_value) {
+    PartialShape expected_shape = {1, {5, 6}};
+
+    auto data = std::make_shared<op::Parameter>(element::f32, pshape_A);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, pshape_B);
+    auto shape_of = make_shared<op::v3::ShapeOf>(target_shape);
+
+    auto op = make_shared<op::v3::Broadcast>(data, shape_of, "NUMPY");
+
+    const auto out_shape = op->get_output_partial_shape(0);
+    EXPECT_EQ(out_shape, expected_shape);
+}
+
+TEST(type_prop, broadcast_v3_labels_in0_dynamic_mixed_dims_numpy) {
+    // Numpy mode for v3::Broadcast mode is one directional
+    // All dimensions of A have labels, B without labels
+    PartialShape pshape_A{-1, 2, 1, -1, {4, 8}, -1, {1, 8}, {1, 10}, {4, 18}};
+    PartialShape pshape_B{-1, 2, {3, 9}, {4, 10}, -1, {5, 11}, -1, {6, 20}, {1, 10}};
+
+    PartialShape expected_shape = {-1, 2, {3, 9}, {4, 10}, -1, {5, 11}, -1, {6, 20}, {1, 10}};
+    // Output shape is a copy of the target shape value, the `A` labels are not propagated
+    std::vector<size_t> expected_labels{0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+    set_shape_labels(pshape_A, {10, 11, 12, 13, 14, 15, 16, 17, 18});
+    set_shape_labels(expected_shape, expected_labels);
+
+    auto data = std::make_shared<op::Parameter>(element::f32, pshape_A);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, pshape_B);
+    auto shape_of = make_shared<op::v3::ShapeOf>(target_shape);
+
+    auto op = make_shared<op::v3::Broadcast>(data, shape_of, "NUMPY");
+
+    const auto out_shape = op->get_output_partial_shape(0);
+
+    EXPECT_EQ(out_shape, expected_shape);
+    EXPECT_EQ(get_shape_labels(out_shape), get_shape_labels(expected_shape));
+}
+
+TEST(type_prop, broadcast_v3_labels_in1_dynamic_mixed_dims_numpy) {
+    // Numpy mode for v3::Broadcast mode is one directional
+    // All dimensions of B have labels, A without labels
+    PartialShape pshape_A{-1, 2, 1, -1, {4, 8}, -1, {1, 8}, {1, 10}, {4, 18}};
+    PartialShape pshape_B{-1, 2, {3, 9}, {4, 10}, -1, {5, 11}, -1, {6, 20}, {1, 10}};
+
+    PartialShape expected_shape = {-1, 2, {3, 9}, {4, 10}, -1, {5, 11}, -1, {6, 20}, {1, 10}};
+    // Output shape is a copy of the target shape, `B` labels are propagated
+    std::vector<size_t> expected_labels{10, 11, 12, 13, 14, 15, 16, 17, 18};
+
+    set_shape_labels(pshape_B, expected_labels);
+    set_shape_labels(expected_shape, expected_labels);
+
+    auto data = std::make_shared<op::Parameter>(element::f32, pshape_A);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, pshape_B);
+    auto shape_of = make_shared<op::v3::ShapeOf>(target_shape);
+
+    auto op = make_shared<op::v3::Broadcast>(data, shape_of, "NUMPY");
+
+    const auto out_shape = op->get_output_partial_shape(0);
+
+    EXPECT_EQ(out_shape, expected_shape);
+    EXPECT_EQ(get_shape_labels(out_shape), get_shape_labels(expected_shape));
+}
+
+TEST(type_prop, broadcast_v3_labels_in0_in1_dynamic_mixed_dims_numpy) {
```

Comment:
Consider refactor test case name as here it uses in0, in1 and in test there are shape A,B.
e.g.
`broadcast_v3_numpy_labels_prop_from_both_inputs`
or something `from_inputs_A_B`.
Check other label propagation tests.